### PR TITLE
Transform Upgrade 1.0 to 1.1

### PR DESCRIPTION
**Pull request by odscjames**

### CODE REVIEW

**Comment by odscjames** [comment]
Again, bits could be improved which if it's ok we'll pick up later
* 1 DB transaction (see comment in code)
* check data actually is 1.0 before trying to transform!

File: ocdskingfisherprocess/cli/commands/new_transform_upgrade_1_0_to_1_1.py [hunk 1]
```diff
@@ -0,0 +1,31 @@
+import ocdskingfisherprocess.database
+import ocdskingfisherprocess.cli.commands.base
+from ocdskingfisherprocess.transform.upgrade_1_0_to_1_1 import Upgrade10To11Transform
+
+
+class NewTransformUpgrade10To11CLICommand(ocdskingfisherprocess.cli.commands.base.CLICommand):
```
Comment (jpmckinney) [inline ocdskingfisherprocess/cli/commands/new_transform_upgrade_1_0_to_1_1.py]: This command is almost identical to the other command. I wonder whether the command should be `new-transform`, which takes a `transform_type` as an argument.

File: ocdskingfisherprocess/transform/upgrade_1_0_to_1_1.py [hunk 2]
```diff
@@ -0,0 +1,83 @@
+from ocdskingfisherprocess.transform.base import BaseTransform
+from ocdskit.upgrade import upgrade_10_11
+import sqlalchemy as sa
+
+
+class Upgrade10To11Transform(BaseTransform):
+    type = 'UPGRADE-1-0-TO-1-1'
```
Comment (jpmckinney) [inline ocdskingfisherprocess/transform/upgrade_1_0_to_1_1.py]: I might prefer an all-lowercase `type` (easier to type) and to change the other transform's `type` slightly to `compile-releases` (make the transform a predicate, not a noun).

File: ocdskingfisherprocess/transform/upgrade_1_0_to_1_1.py [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+
+    def process(self):
```
Comment (jpmckinney) [inline ocdskingfisherprocess/transform/upgrade_1_0_to_1_1.py]: I'm not sure why `process_all_files` should be a separate method from `process`, as they do the same thing.

File: ocdskingfisherprocess/transform/upgrade_1_0_to_1_1.py [hunk 4]
```diff
@@ -0,0 +1,83 @@
...
+            release_rows = connection.execute(
+                self.database.release_table.select().where(
+                    self.database.release_table.c.collection_file_item_id == file_item_model.database_id)
+            )
+
+        for release_row in release_rows:
+            if not self.has_release_id_been_done(release_row['id']):
+                self.process_release_row(file_model, file_item_model, release_row)
+
+        del release_rows
```
Comment (jpmckinney) [inline ocdskingfisherprocess/transform/upgrade_1_0_to_1_1.py]: Wouldn't it be clearer to use SQLAlchemy's `.close()` function?

**Comment by jpmckinney** [comment]
> check data actually is 1.0 before trying to transform!

The OCDS Kit method does this already. If it's not 1.0, it does nothing.

File: requirements.in [hunk 5]
```diff
@@ -1,5 +1,7 @@
 -e git+https://github.com/OpenDataServices/flatten-tool.git@v0.5.0#egg=flattentool
 -e git+https://github.com/open-contracting/lib-cove-ocds.git@v0.1.0#egg=libcoveocds
+# After version 0.0.5 of https://pypi.org/project/ocdskit/ has been released, we can just switch to that
```
Comment (jpmckinney) [inline requirements.in]: I've now released 0.0.5

Comment (odscjames) [reply requirements.in]: Thank you

**Comment by odscjames** [comment]
> > check data actually is 1.0 before trying to transform!
> 
> The OCDS Kit method does this already. If it's not 1.0, it does nothing.

I would still like to do this at the Kingfisher level
A) We can avoid rewriting the same data back to the database, which will both be faster and save some space (or maybe we still want that data in the new collection? I'm thinking about what the case would be if a collection had both 1.0 and 1.1 data, tho that sounds unlikely. Whatever, I'd like to consciously discuss and make sure we do the right thing. ) 
B) We can log that it wasn't 1.0, and then provide a clearer status report to the user. "There was no 1.0 data to transform" as a message immediately informs the user and means they don't waste time trying to run analysis against data that is actually exactly the same.

File: ocdskingfisherprocess/transform/upgrade_1_0_to_1_1.py (hunk 4 above)
Comment (odscjames) [reply ocdskingfisherprocess/transform/upgrade_1_0_to_1_1.py]: Whatever the right thing to do is, it should also be done to the checks code. So I'll make an issue to look into this, and do the right thing to both. Thanks.

File: ocdskingfisherprocess/transform/upgrade_1_0_to_1_1.py (hunk 3 above)
Comment (odscjames) [reply ocdskingfisherprocess/transform/upgrade_1_0_to_1_1.py]: Yes. fixed.

File: ocdskingfisherprocess/cli/commands/new_transform_upgrade_1_0_to_1_1.py (hunk 1 above)
Comment (odscjames) [reply ocdskingfisherprocess/cli/commands/new_transform_upgrade_1_0_to_1_1.py]: Is it ok to come back to this? As said in last P.R., I do want to look at how it's triggered anyway - we have already discussed user needs around this being triggered by default, for instance.

File: ocdskingfisherprocess/transform/upgrade_1_0_to_1_1.py (hunk 2 above)
Comment (odscjames) [reply ocdskingfisherprocess/transform/upgrade_1_0_to_1_1.py]: >  might prefer an all-lowercase type (easier to type) 

I don't see any case where people should be typing these directly, and in any case it should be easy to correct case.

> and to change the other transform's type slightly to compile-releases (make the transform a predicate, not a noun).

Ahh, I have mixed up language here. I'll change 'COMPILED-RELEASES' to 'COMPILE-RELEASES' (and associated stuff) in another small pull request. Thanks.

File: ocdskingfisherprocess/cli/commands/new_transform_upgrade_1_0_to_1_1.py (hunk 1 above)
Comment (jpmckinney) [reply ocdskingfisherprocess/cli/commands/new_transform_upgrade_1_0_to_1_1.py]: Sure, let's make sure we come back to it.

**Pull request merged**